src/main/java/seedu/address/ui/control/PopupAddInternship.java

```diff
@@ -0,0 +1,328 @@
+            TextField rf = (TextField) reflectionVBox.getChildren().get(i);
+            if (rf.getText().matches(ILLEGAL_REGEX)) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    /**
+     * Handles the add internship button clicked event.
+     */
+    @FXML
+    private void handleAddInternship() {
+        if (!isValid()) {
+            return;
+        }
+
+        String commandText = AddCommand.COMMAND_WORD
+                + " " + CliSyntax.PREFIX_COMPANY_NAME.getPrefix()
+                + companyName.getText()
+                + " " + CliSyntax.PREFIX_JOB_TITLE.getPrefix()
+                + jobTitle.getText();
+
+        if (!place.getText().isBlank()) {
+            commandText += " " + CliSyntax.PREFIX_LOCATION.getPrefix() + place.getText();
+        }
+
+        if (!salary.getText().isBlank()) {
+            commandText += " " + CliSyntax.PREFIX_SALARY.getPrefix() + salary.getText();
+        }
+
+        if (!rating.getText().isBlank()) {
+            commandText += " " + CliSyntax.PREFIX_RATING.getPrefix() + rating.getText();
+        }
+
+        for (int i = 1; i < qualificationVBox.getChildren().size(); i += 1) {
+            TextField q = (TextField) qualificationVBox.getChildren().get(i);
+            if (!q.getText().isBlank()) {
+                commandText += " " + CliSyntax.PREFIX_QUALIFICATION.getPrefix() + q.getText();
+            }
+        }
+
+        for (int i = 1; i < programmingLanguageVBox.getChildren().size(); i += 1) {
+            TextField pl = (TextField) programmingLanguageVBox.getChildren().get(i);
+            if (!pl.getText().isBlank()) {
+                commandText += " " + CliSyntax.PREFIX_PROGRAMMING_LANGUAGE.getPrefix() + pl.getText();
+            }
+        }
+
+        for (int i = 1; i < reviewVBox.getChildren().size(); i += 1) {
+            TextField r = (TextField) reviewVBox.getChildren().get(i);
+            if (!r.getText().isBlank()) {
+                commandText += " " + CliSyntax.PREFIX_REVIEW.getPrefix() + r.getText();
+            }
+        }
+
+        for (int i = 1; i < noteVBox.getChildren().size(); i += 1) {
+            TextField n = (TextField) noteVBox.getChildren().get(i);
+            if (!n.getText().isBlank()) {
+                commandText += " " + CliSyntax.PREFIX_NOTE.getPrefix() + n.getText();
+            }
+        }
+
+        for (int i = 1; i < reflectionVBox.getChildren().size(); i += 1) {
+            TextField rf = (TextField) reflectionVBox.getChildren().get(i);
+            if (!rf.getText().isBlank()) {
+                commandText += " " + CliSyntax.PREFIX_REFLECTION.getPrefix() + rf.getText();
+            }
+        }
+
+        try {
+            mainWindow.executeCommand(commandText);
+        } catch (CommandException e) {
+            e.printStackTrace();
+        } catch (ParseException e) {
+            e.printStackTrace();
+        } finally {
+            hide();
+        }
+    }
+
```

Comment:
This method also seems too long to me. It may be better if the code is short and simple.